ROADMAP.md

```diff
@@ -0,0 +1,61 @@
+# Roadmap for Packer v2
+
+`packer` has become bloated, compilation (at least as it currently stands) has proven confusing and probably not necessary, and the codebase has become messy and hard to maintain.
+
+As such, I'm proposing/working on (very slowly, as my OSS time is quite limited in my current job) an incremental rewrite of `packer`.
+The below is a high-level overview of this plan.
+Comments and additions are welcome and encouraged!
+
+## General principles
+
+`packer`'s code currently suffers from some poor software development practices largely stemming from getting "discovered" before it had time to become polished.
+We want to avoid these pitfalls in the rewrite:
+
+- Document all functions with emmylua-style comments: `packer` has very sparse documentation in its codebase. New/updated functions need to be appropriately documented.
+- Add tests where possible: testing a package manager can be a pain because of requirements to interact with the file system, etc. However, `packer`'s current tests do not cover much, and new code should at least strongly consider adding unit tests.
+- Avoid giant functions: `packer`'s logic contains some monstrous functions (e.g., `manage`, the main compilation logic, etc.) that are difficult to work with. New code should strive to use more, simpler functions in a more modular design.
+- Prioritize clean, performant code over total flexibility: `packer`'s design allows for a very flexible range of input formats. Although we don't want to completely lose this property, supporting all of these formats (and other use cases) has contributed to code bloat. If there's a choice to be made between losing some flexibility and significantly increasing the complexity of the code, seriously consider removing the complexity.
+- Reduce redundancy: `packer` currently has an annoying amount of redundant/overlapping functionality (the worst offenders are the `requires`/`wants`/`after` family of keywords, but there's also duplication of logic in the utilities, etc.). The rewrite should aim to reduce this.
```

Comment:
Generally this looks great!

Regarding redundancy, there's a lot of code-duplication in `packer.update` and `packer.sync`. Maybe there are reasons for this but if things are structured well, it seems like `sync` could just be something like:
```lua
packer.sync = function(..)
  packer.clean()
  packer.update(...)
  packer.compile()
end
```

Reply:
That's the idea! I just haven't gotten to that part of the rewrite yet.